src/codebert/codebert_model.rs

```diff
@@ -0,0 +1,360 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::bert::{BertConfig, BertModel};
+use crate::common::error::RustBertError;
+use crate::pipelines::common::TokenizerOption;
+use crate::pipelines::sequence_classification::SequenceClassificationConfig;
+use crate::resources::ResourceProvider;
+use crate::roberta::{
+    RobertaEmbeddings, RobertaForMaskedLM, RobertaForMultipleChoice, RobertaForQuestionAnswering,
+    RobertaForSequenceClassification, RobertaForTokenClassification,
+};
+use crate::Config;
+use rust_tokenizers::tokenizer::TruncationStrategy;
+use rust_tokenizers::TokenizedInput;
+use std::borrow::Borrow;
+use tch::nn::VarStore;
+use tch::{nn, no_grad, Device, Tensor};
+
+/// # CODEBERT Pretrained model weight files
+pub struct CodeBertModelResources;
+
+/// # CODEBERT Pretrained model config files
+pub struct CodeBertConfigResources;
+
+/// # CODEBERT Pretrained model vocab files
+pub struct CodeBertVocabResources;
+
+/// # CODEBERT Pretrained model merges files
+pub struct CodeBertMergesResources;
+/// CODEBERT has same basic module implementation as ROBERTA
+pub type CodeBertConfig = BertConfig;
+pub type CodeBertForFeatureExtractionConfig = SequenceClassificationConfig;
+pub type CodeBertEmbeddings = RobertaEmbeddings;
+pub type CodeBertForMaskedLM = RobertaForMaskedLM;
+pub type CodeBertForMultipleChoice = RobertaForMultipleChoice;
+pub type CodeBertForSequenceClassification = RobertaForSequenceClassification;
+pub type CodeBertForSentenceEmbeddings = BertModel<CodeBertEmbeddings>;
+pub type CodeBertForTokenClassification = RobertaForTokenClassification;
+pub type CodeBertForQuestionAnswering = RobertaForQuestionAnswering;
+impl CodeBertModelResources {
+    /// Shared under Apache 2.0 license by the Microsoft team at <https://github.com/microsoft/CodeBERT>. Modified with conversion to C-array format.
+    pub const CODEBERT: (&'static str, &'static str) = (
+        "codebert/model",
+        "https://huggingface.co/microsoft/codebert-base/resolve/main/rust_model.ot",
+    );
+    /// Shared under Apache 2.0 license by the Hugging Face Inc. team at <https://huggingface.co/huggingface/CodeBERTa-language-id>. Modified with conversion to C-array format.
+    pub const CODEBERTA_LANG: (&'static str, &'static str) = (
+        "codeberta-language-id/model",
+        "https://huggingface.co/huggingface/CodeBERTa-language-id/resolve/main/rust_model.ot",
+    );
+    /// Shared under Apache 2.0 license by the Microsoft team at <https://github.com/microsoft/CodeBERT>. Modified with conversion to C-array format.
+    pub const CODEBERT_MLM: (&'static str, &'static str) = (
+        "codebert-mlm/model",
+        "https://huggingface.co/microsoft/codebert-base-mlm/resolve/main/rust_model.ot",
+    );
+}
+
+impl CodeBertConfigResources {
```

Comment:
The easiest option would be to add these additional resources to https://github.com/guillaume-be/rust-bert/blob/59c0e668d68eb80c674100049ecd37c35c9ab4b5/src/roberta/roberta_model.rs#L35 - it seems there are no architectural differences between CodeBERT and RoBERTa.